html_css/intermediate_html/form-validations.md

```diff
@@ -113,7 +113,15 @@ Try submitting the form with seven passengers to see the validation in action.
 To ensure we get the correct information from users, we will often want to ensure data matches a particular pattern.
 Real-world applications would be checking a credit card number or a zipcode is in the correct format.
 
-To add a pattern validation, we give the form control a `pattern` attribute with a [regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) as the value. In our example we are using the pattern validation to ensure a US zip code is in the correct format. 5 numbers followed by an optional dash and 4 more numbers:
+To add a pattern validation, we give the form control a `pattern` attribute with a [regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) as the value. In our example we are using the pattern validation to ensure a US zip code is in the correct format. 5 numbers followed by an optional dash and 4 more numbers: Using the Regular expression pattern `(\\d{5}([\\-]\\d{4})?)`. How exactly do we interpret this expression? Let’s take this step by step, going from left to right:
+<ul> 
+  <li> Capturing group: ( ) </li>
+  <li> Escape character: \ </li>
+  <li> Digit: \d </li>
+  <li> Number of characters: { } </li>
+  <li> Matches characters in brackets: [ ] </li>
+  <li> 0 or 1 character: ? </li>
+</ul>
```

Comment:
I don't think we should include this information at all. What is stated here is already covered in the link you provided, and this doesn't really help explain *what* exactly the characters in the regex are doing. Revert this back to the original text:

```suggestion
To add a pattern validation, we give the form control a `pattern` attribute with a [regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) as the value. In our example we are using the pattern validation to ensure a US zip code is in the correct format. 5 numbers followed by an optional dash and 4 more numbers:
```